### PR TITLE
Chore/link backend to executor

### PR DESCRIPTION
**Pull request by limanjun99**
This PR adds (very simple) functionality to the `POST /execute` endpoint of the backend API, by linking it to the executor Lambda.

- Added input validation for HTTP requests with celebrate and joi
- Updated documentation for backend to use one swagger.yml file instead of swagger-jsdoc. This is so that we can use the [swagger editor](https://editor.swagger.io/)
- Gave `lambda:InvokeFunction` permissions to the backend Lambda, allowing it to invoke the executor Lambda

Screenshot 2023-05-25 at 6 14 27 PM

Next steps:
- Investigate how the Python debugger behaves with various constructs like functions or input/output
- Investigate possible errors thrown and handle them appropriately

### CODE REVIEW

File: backend/env/.env.development [hunk 1]
```diff
@@ -1,2 +1,4 @@
 NODE_ENV=development
-PORT=8000
\ No newline at end of file
+PORT=8000
+EXECUTOR_REGION=us-east-1
```
Comment (huajun07) [inline backend/env/.env.development:3]: Is there a reason we are using `us-east-1` instead of `ap-southeast-1`?

File: backend/serverless.yml [hunk 2]
```diff
@@ -1,19 +1,25 @@
 service: backend
 frameworkVersion: '3'
 
-useDotenv: true
 plugins:
   - serverless-plugin-typescript
+  - serverless-dotenv-plugin
```
Comment (huajun07) [inline backend/serverless.yml:6]: Do take note that using `serverless-dotenv-plugin` also deploys the envars to the lambda. Not sure if that's what you intended

File: backend/env/.env.development (hunk 1 above)
Comment (limanjun99) [reply backend/env/.env.development]: No reason, it is the default region chosen by Serverless. Updated to deploy to ap-southeast-1 (Singapore) instead.

File: backend/serverless.yml (hunk 2 above)
Comment (limanjun99) [reply backend/serverless.yml:6]: Yup, all the .env files are for non-secret configuration, secrets will be handled by other methods (like Github Secrets)

File: backend/src/execute/execute.service.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+export default async function execute(code: string) {
+	const command = new InvokeCommand({
+		FunctionName: config.get('executorName'),
+		Payload: Buffer.from(JSON.stringify(code)),
+	})
+	const { Payload } = await client.send(command)
+	if (Payload === undefined) {
+		return null
+	}
+	const result = JSON.parse(Buffer.from(Payload).toString())
```
[the remark this answers is not in the document]
Comment (limanjun99) [reply backend/src/execute/execute.service.ts:15]: Will be added later on after checking out what kind of errors/responses can be returned from the executor

[the remark this answers is not in the document]
Comment (huajun07) [reply backend/src/execute/execute.service.ts:15]: Ok, please make a task for this

**Review by huajun07 — APPROVED** [review]

**Pull request merged**